devcluster/stage.py

```diff
@@ -252,10 +252,12 @@ def kill(self, sig: typing.Optional[signal.Signals] = None) -> None:
         # kill via signal
         self.dying = True
         assert self.proc
-        if sig is None:
-            self.proc.kill()
-        else:
+        if sig is not None:
             self.proc.send_signal(sig)
+        elif self.config.kill_signal is not None:
```

Comment:
hm, I don't remember exactly how the existing kill signal gets configured, but I think it should be set by whoever calls `kill(sig=...)`.

I think pulling from the config at this point might be creating some confusing duplicate codepaths.

"confusing codepaths in devcluster"... lol

Reply:
My potentially incorrect understanding is the signal parameter of ```kill_stage``` can be set from when interacting with devcluster from Python code as we do in managed devcluster tests

https://github.com/determined-ai/determined/blob/d13169c98b43d4c17cd94736102ecdf0e9ddb638/e2e_tests/tests/cluster/managed_cluster.py#L52 (though we don't actually set it in this case)

I think the docker stage ```kill_signal``` also uses the signal parameter if exists and pulls from config if None? 

https://github.com/determined-ai/devcluster/blob/3aac7871ea83e9d37efbef23bfaa9f1dec05b4e3/devcluster/stage.py#L456

Reply:
ah, that all looks right.  Great work.  Sorry about that.